File: Pertemuan-05-Autonics-Pemanas-Air-Manual/examples/analyze_manual_response.m
clear; clc; close all;
file=fullfile('..','templates','template_pengamatan.csv');
T=readtable(file);
valid=~isnan(T.temperature_C); T=T(valid,:);
if height(T)<3, error('Isi template dengan minimal 3 data temperatur.'); end
figure; plot(T.time_s,T.temperature_C,'o-'); hold on; plot(T.time_s,T.setpoint_C,'--'); grid on;
xlabel('Time (s)'); ylabel('Temperature (C)'); legend('PV','SV');
target=median(T.setpoint_C(end-max(1,round(height(T)*0.1)):end));
y0=T.temperature_C(1); A=target-y0;
peak=max(T.temperature_C); overshoot=max(0,(peak-target)/max(abs(A),eps)*100);
ess=target-mean(T.temperature_C(max(1,end-2):end));
fprintf('Overshoot = %.2f %%\nSteady-state error = %.3f C\n',overshoot,ess);

File: Pertemuan-05-Autonics-Pemanas-Air-Manual/examples/analyze_manual_response.py
import pandas as pd, matplotlib.pyplot as plt
from pathlib import Path
import sys
sys.path.append(str(Path(__file__).resolve().parents[2]/'shared'))
from control_metrics import step_metrics
f=Path(__file__).resolve().parents[1]/'templates'/'template_pengamatan.csv'
df=pd.read_csv(f).dropna(subset=['temperature_C'])
if len(df)<3: raise SystemExit('Isi template dengan minimal 3 data.')
m=step_metrics(df.time_s,df.temperature_C,df.setpoint_C)
print(m)
plt.plot(df.time_s,df.temperature_C,'o-',label='PV'); plt.plot(df.time_s,df.setpoint_C,'--',label='SV'); plt.grid(); plt.legend(); plt.show()
